packages/admin/admin/src/common/FieldSet.tsx

```diff
@@ -0,0 +1,149 @@
+    title: React.ReactNode;
+    supportText?: React.ReactNode;
+    endAdornment?: React.ReactNode;
+    collapsible?: boolean;
+    initiallyExpanded?: boolean;
+    disablePadding?: boolean;
+    componentsProps?: FieldSetComponentsProps;
+}
+
+export type FieldSetClassKey = "root" | "headerColumn" | "title" | "supportText" | "endAdornment" | "placeholder" | "children" | "disablePadding";
+
+const styles = (theme: Theme) =>
+    createStyles<FieldSetClassKey, FieldSetProps>({
+        root: {
+            display: "flex",
+            flexDirection: "row-reverse",
+            padding: "0 10px",
+            height: "80px",
+            [theme.breakpoints.up("sm")]: {
+                padding: "0 20px",
+            },
+        },
+        headerColumn: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            alignContent: "center",
+            padding: "0px",
+            [theme.breakpoints.up("sm")]: {
+                padding: "10px",
+            },
+        },
+        title: {
+            display: "flex",
+            alignItems: "center",
+            fontWeight: theme.typography.fontWeightMedium,
+            fontSize: "16px",
+            textTransform: "uppercase",
+            color: theme.palette.text.primary,
+        },
+        supportText: {
+            fontSize: "12px",
+            lineHeight: "18px",
+            color: theme.palette.text.secondary,
+        },
+        endAdornment: { display: "flex", alignItems: "center" },
+        placeholder: {
+            flexGrow: 1,
+            boxSizing: "inherit",
+            userSelect: "none",
+        },
+        children: {
+            display: "flex",
+            flexDirection: "column",
+            borderTop: `1px solid ${theme.palette.divider}`,
+            padding: "20px",
+            [theme.breakpoints.up("sm")]: {
+                padding: "40px",
+            },
+            "&$disablePadding": {
+                padding: "0px",
+            },
+        },
+        disablePadding: {},
+    });
+
+function FieldSet({
+    title,
+    supportText,
+    endAdornment,
+    children,
+    collapsible = true,
+    initiallyExpanded = true,
+    disablePadding = false,
+    componentsProps,
+    classes,
+}: React.PropsWithChildren<FieldSetProps> & WithStyles<typeof styles>): React.ReactElement {
+    const [expanded, setExpanded] = React.useState(initiallyExpanded);
+
+    const handleChange = (event: React.SyntheticEvent, isExpanded: boolean) => {
+        setExpanded(isExpanded);
+    };
+
+    return (
+        <MuiAccordion
+            expanded={expanded}
+            onChange={
+                collapsible
+                    ? handleChange
+                    : () => {
+                          /* do nothing */
+                      }
+            }
+            {...componentsProps?.root}
+        >
+            <MuiAccordionSummary
+                classes={{ root: classes.root }}
```

Comment:
The classes used in this component are confusing: root, which should be used for the root of the component (MuiAccordion), is instead used for the summary, which should be summary instead. @jamesricky can you please help @NataliaVizintini to correctly name the classes?